packages/medusa/src/strategies/batch-jobs/price-list/import.ts

```diff
@@ -131,6 +150,64 @@ class PriceListImportStrategy extends AbstractBatchJobStrategy {
 
     const pricesToCreate: PriceListImportOperation[] = []
 
+    const invalidVariantIds: string[] = []
+    const invalidVariantSkus: string[] = []
+
+    const csvDataRowIds = [
+      ...new Set(csvData.map((row) => `${row[PriceListRowKeys.VARIANT_ID]}`)),
+    ]
+    const csvDataRowSkus = [
+      ...new Set(csvData.map((row) => `${row[PriceListRowKeys.VARIANT_SKU]}`)),
+    ]
+
+    let idVariants: {
+      id: string
+    }[] = []
+    let skuVariants: {
+      id: string
+      sku: string | null
+    }[] = []
+    if (this.featureFlagRouter_.isFeatureEnabled(MedusaV2Flag.key)) {
+      const idQuery = {
+        variants: {
+          __args: {
+            id: csvDataRowIds,
+          },
+          fields: ["id"],
+        },
+      }
+
+      idVariants = await this.remoteQuery(idQuery)
+
+      const skuQuery = {
+        variants: {
+          __args: {
+            sku: csvDataRowSkus,
+          },
+          fields: ["id", "sku"],
+        },
+      }
+
+      skuVariants = await this.remoteQuery(skuQuery)
+    } else {
+      idVariants = await this.productVariantService_.list(
+        {
+          id: csvDataRowIds,
+        },
+        { select: ["id"] }
+      )
+
+      skuVariants = await this.productVariantService_.list(
+        {
+          sku: csvDataRowSkus,
+        },
+        { select: ["id", "sku"] }
+      )
+    }
```

Comment:
todo: single call with or selector. if needed, we can discuss it. I ve done something similar for the update of the line items then you can loop and do what is necessary